### PR TITLE
Product Detail: Add intrumentation tests

### LINKED ISSUE #1043: Add instrumentation tests for Product detail page
### Product List - Expanded
- [x] Clicking on the `DETAILS` button redirects to expanded version of product list screen.
- [x]  testing currency formatter for multiple currencies (dollar, pound, inr)
- [x]  if sku not empty, verify it is displayed
- [x]  if sku is empty, verify it is not displayed
- [x]  Shipping and taxes is displayed correctly.

### Product Detail Page
- [x] Clicking on the a product item redirects to product detail page
- [x] The product image, product title, total orders for the product and the reviews for the product is displayed correctly.
- [x] Clicking on the `View product on store` button should redirect to the product page inside webview.
- [x] Testing pricing info for the product for multiple currencies
- [x] Testing inventory of product is displayed correctly

### Product Detail Navigation
- [x] Clicking on `X` button redirects to Product list page.
- [x] When scrolling to bottom of product detail page, the title is set to the name of the product.
- [x] Clicking on `Share` button opens intent to share the product

### PR DESCRIPTION
**Pull request by anitaa1990**
Fixes #1043 by adding UI tests for Product list and product detail page.

The following scenarios have been added to the test cases:
### Product List - Expanded
- [x] Clicking on `Details` button in Order detail page, redirects to the expanded Product List page.
- [x] Product Item: The product name, quantity, total, tax and their labels are displayed correctly
- [x] Product Item: Currency formatting for product prices, taxes are displayed correctly for US, EUR, IND, AUD
- [x] SKU is displayed only id available and hidden, if not.

#### To Test
- [x] Run `ProductListCardTest`

### Product Detail Page
- [x] Clicking on the a product list item redirects to Product Detail page.
- [x] The toolbar title, the `X` icon and share button are displayed correctly.
- [x] Clicking on `X` icon redirects to the previous screen.
- [x] Clicking on `Share` button opens intent to share the product
- [x] Product Detail: Image is only displayed, if available
- [x] Product Detail: The product title, total orders, review labels are displayed correctly.
- [x] verify the toolbar title is displayed correctly for different product types
- [x] verify that total orders is formatted correctly for different order totals. For instance: 
  - [x] 200 -> 200
  - [x] 2000 -> 2k
  - [x] 20000 -> 20k
  - [x] 2000000 - > 2m
- [x] Check that `Product view on store button` is displayed correctly
- [x] Verify that clicking on the button opens WebView.
- [x] verify that status badge is displayed for unpublished products
- [x] verify that reviews are only displayed if allowed

#### Variation Products:
- [x] Verify that variations are not displayed for variation products
- [x] Verify that reviews are not displayed for variation products

#### Pricing and Inventory card:
- [x] Pricing card title is displayed correctly. If we have pricing info this card is "Pricing and inventory" otherwise it's just "Inventory".
- [x] When there's a sale price show price & sales price as a group, otherwise show price separately. 
- [x] show stock properties as a group if stock management is enabled, otherwise show sku separately.

#### Purchase Detail card:
- [x] Verify shipping info displayed correctly when width/height/length is not empty
- [x] Verify shipping info handled correctly when weight/height is not empty
- [x] Verify shipping title is displayed correctly
- [x] Verify weight, size, shipping class displayed correctly
- [x] Verify pricing Info is displayed correctly when sale price is available
- [x] Verify downloadable products are displayed correctly
- [x] Verify downloadable products are displayed correctly without limit or expiry
- [x] verify purchase note is displayed correctly, only if available
- [x] verify a really long purchase note opens a dialog when `Read More` button is clicked

#### To test:
- [x] Run `ProductDetailNavigationTest`

### CODE REVIEW

File: WooCommerce/src/androidTest/kotlin/com/woocommerce/android/ui/orders/WcOrderTestUtils.kt [hunk 1]
```diff
@@ -417,6 +428,42 @@ object WcOrderTestUtils {
                 currencyDecimalNumber = 2)
     }
 
+    /**
+     * Generates a single [WCProductSettingsModel] object for a mock Site
+     */
+    fun generateProductSettings(
```
Comment (nbradbury) [inline WooCommerce/src/androidTest/kotlin/com/woocommerce/android/ui/orders/WcOrderTestUtils.kt]: @anitaa1990 I'm curious why these product-related functions are in an order-related class?

Comment (anitaa1990) [reply WooCommerce/src/androidTest/kotlin/com/woocommerce/android/ui/orders/WcOrderTestUtils.kt]: Ah, that's a good point! Thanks, I didn't notice that! I started testing with Orders so I think I dumped them all into the same class. I've modified the Product settings and `WcProductModel` into a separate class. I also noticed other methods in `WCOrderTestUtils` that are not specific to Orders. So I moved them into their own class too. Both done in 2e007b0 👍

**Comment by franc3sc0s92** [comment]
hi, can add to the app the possibility to show other field of the product?

**Comment by AmandaRiu** [comment]
@franc3sc0s92 Which field are you referring to?

**Pull request merged**